src/convertToObject.js

```diff
@@ -3,16 +3,32 @@
 /**
  * Implement convertToObject function:
  *
- * Function takes a string with styles (see an example in [stylesString.js](./stylesString.js))
+ * Function takes a string with styles
+ * (see an example in [stylesString.js](./stylesString.js))
  * and returns an object where CSS properties are keys
- * and values are the values of related CSS properties (see an exampl in [test file](./convertToObject.test.js))
+ * and values are the values of related CSS properties
+ * (see an exampl in [test file](./convertToObject.test.js))
  *
  * @param {string} sourceString
  *
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const resultObject = sourceString.split(';')
+    .map(x => x.split(':').map(y => y.trim()))
+    .reduce((a, x) => {
+      a[x[0]] = x[1];
+
+      return a;
+    }, {});
+
+  for (const item in resultObject) {
+    if (!item.length) {
+      delete resultObject[item];
+    }
+  }
```

Comment:
```suggestion
    .map(x => x.split(':')
    .reduce((a, x) => {
      a[x[0].trim()] = x[1].trim();

      return a;
    }, {});
  }
```
I guess it will work

Reply:
it doesn't work :(
![Знімок екрана 2022-12-12 о 13 05 13](https://user-images.githubusercontent.com/112945779/207032859-0e7100c8-16b8-4786-aa81-c037d16a4249.png)
![Знімок екрана 2022-12-12 о 13 05 58](https://user-images.githubusercontent.com/112945779/207032865-b37f488f-473a-46bb-9d4f-9977addcd892.png)
![Знімок екрана 2022-12-12 о 13 06 26](https://user-images.githubusercontent.com/112945779/207032867-d13e860c-940b-40e4-a92f-722981525b40.png)

Reply:
I made it in the other way